javalib/src/main/scala/java/util/Objects.scala

```diff
@@ -0,0 +1,64 @@
+package java.util
+
+object Objects {
+
+  @inline
+  def equals(a: AnyRef, b: AnyRef): Boolean =
+    if (a == null) b == null
+    else a.equals(b)
+
+  @inline
+  def deepEquals(a: AnyRef, b: AnyRef) = {
+    (a, b) match {
+      case (a: Array[AnyRef], b: Array[AnyRef]) => Arrays.deepEquals(a, b)
+      case _                                    => equals(a, b)
+    }
+  }
+
+  @inline
+  def hashCode(o: AnyRef) =
+    if (o == null) 0
+    else o.hashCode()
+
+  @inline
+  def hash(values: Array[AnyRef]): Int =
+    Arrays.hashCode(values)
+
+  @inline
+  def toString(o: AnyRef): String =
+    String.valueOf(o)
+
+  @inline
+  def toString(o: AnyRef, nullDefault: String): String =
+    if (o == null) nullDefault
+    else o.toString
+
+  @inline
+  def compare[T](a: T, b: T, c: Comparator[_ >: T]): Int =
+    if (a == b) 0
+    else c.compare(a, b)
```

Comment:
`a eq b`. The spec is unclear about what "identical" means, but experimentation clearly shows that it means `eq`:
```scala
scala> Objects.compare((1, 2), (1, 2), new java.util.Comparator[AnyRef] { def compare(x: AnyRef, y: AnyRef) = 1 })
res3: Int = 1
```